Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,472 @@
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Finds the current balance of a single person in array.
+        /// </summary>
+        private static decimal CurrentBalanceForSinglePerson(string personAndBalance)
+        {
+            var balanceList = ConvertStringToDecimalArray(personAndBalance);
+            try
+            {
+                decimal currentBalance = balanceList[^1];
+
+                return currentBalance;
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine(ex.Message);
+            }
+            return default;
+        }
+
+        /// <summary>
+        /// Return index of Max value in decimal array.
+        /// </summary>
+        private static int FindDecimalArrayMax(decimal[] inputArray)
+        {
+            var max = decimal.Zero;
+            try
+            {
+                max = inputArray[0];
+            }
+            catch (IndexOutOfRangeException ex)
+            {
+                Debug.WriteLine(ex.Message);
+            }
+            var index = 0;
+            for (int i = index; i < inputArray.Length; i++)
+            {
+                if (inputArray[i] > max)
+                {
+                    max = inputArray[i];
+                    index = i;
+                }
+            }
+            return index;
+        }
+
+        /// <summary>
+        /// Return index of Min value in decimal array.
+        /// </summary>
+        private static int FindDecimalArrayMin(decimal[] inputArray)
+        {
+            var min = decimal.Zero;
+
+            try
+            {
+                min = inputArray[0];
+            }
+            catch (IndexOutOfRangeException ex)
+            {
+                Debug.WriteLine(ex.Message);
+            }
+            var index = 0;
+            for (int i = 0; i < inputArray.Length; i++)
+            {
+                if (inputArray[i] < min)
+                {
+                    min = inputArray[i];
+                    index = i;
+                }
+            }
+            return index;
+        }
+
+        /// <summary>
+        /// Returns an formated output of currency (ex. -¤1, ¤4, ¤1002, -¤1001).
+        /// </summary>
+        private static string FormatCurrency(decimal currency)
         {
-            return "";
+            const char negativeSymbol = '-';
+
+            var formatedCurrency = new StringBuilder();
+
+            if (currency < 0)
+            {
+                currency *= -1;
+                if (defaultCurrencyLocation == 0)
+                    formatedCurrency.Append(negativeSymbol).Append(defaultCurrencySymbol).Append(currency);
+                if (defaultCurrencyLocation == 1)
+                    formatedCurrency.Append(defaultCurrencySymbol).Append(" ").Append(negativeSymbol).Append(currency);
+                return formatedCurrency.ToString();
+            }
+
+            if (defaultCurrencyLocation == 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4971928)